src/lightning/fabric/accelerators/xla.py

```diff
@@ -99,12 +100,8 @@ def is_available() -> bool:
         return queue.get_nowait()
 
     @classmethod
-    def register_accelerators(cls, accelerator_registry: Dict) -> None:
-        accelerator_registry.register(
-            "tpu",
-            cls,
-            description=cls.__class__.__name__,
-        )
+    def register_accelerators(cls, accelerator_registry: _AcceleratorRegistry) -> None:
+        accelerator_registry.register("tpu", cls, description=cls.__class__.__name__)
```

Comment:
To complete #16130, we'll want that `accelerator="tpu"` explicitly requires that a TPU device is available whereas `accelerator="xla"` can be used with a CUDA device.

Similar to the difference between `accelerator="cuda"` and `accelerator="gpu"`